enhancements/network/user-defined-network-segmentation.md

```diff
@@ -0,0 +1,947 @@
+IP on the cluster default network instead.
+* Admin ability to configure networks to have access to all services and/or expose services to be accessible from all
+networks.
+* Ability to advertise user-defined networks to external networks using BGP/EVPN. This will enable things like:
+    * External -> Pod ingress per VRF (Ingress directly to pod IP)
+    * Multiple External Gateway (MEG) in a BGP context, with ECMP routes
+* Allow connection of multiple networks via explicit router API configuration.
+
+## Proposal
+
+By default in OVN-Kubernetes, pods are attached to what is known as the “cluster default" network, which is a routed network
+divided up into a subnet per node. All pods will continue to have an attachment to this network, even when assigned a
+different primary network. Therefore, when a pod is assigned to a user-defined network, it will have two interfaces, one
+to the cluster default network, and one to the user-defined network. The cluster default network is required in order to provide:
+
+1. KAPI service access
+2. DNS service access
+3. Kubelet healthcheck probes to the pod
+4. User defined ports that will be exposed on the cluster default network. This may be used for things like prometheus
+   metric scraping.
+
+All other traffic from the pod will be dropped by firewall rules on this network, when the pod is assigned a user-defined
+primary network. Routes will be added to the pod to route KAPI/DNS traffic out towards the cluster default network. Note,
+it may be desired to allow access to any Kubernetes service on the cluster default network (instead of just KAPI/DNS),
+but at a minimum KAPI/DNS will be accessible. Furthermore, the IP of the pod from the Kubernetes API will continue to
+show the IP assigned in the cluster default network.
+
+In OVN-Kubernetes secondary networks are defined using Network Attachment Definitions (NADs). For more information on
+how these are configured, refer to:
+
+[https://github.com/ovn-org/ovn-kubernetes/blob/master/docs/features/multi-homing.md](https://github.com/ovn-org/ovn-kubernetes/blob/master/docs/features/multi-homing.md)
+
+The proposal here is to leverage this existing mechanism to create the network. The only new field that will be
+introduced to the NAD spec is a “primaryNetwork” key which indicates that this network should be used for the pods
+primary network. For example, an OVN-Kubernetes NAD may look like:
+
+```
+apiVersion: k8s.cni.cncf.io/v1
+kind: NetworkAttachmentDefinition
+metadata:
+  name: l3-network
+  namespace: ns1
+spec:
+  config: |2
+    {
+            "cniVersion": "0.3.1",
+            "name": "l3-network",
+            "type": "ovn-k8s-cni-overlay",
+            "topology":"layer3",
+            "subnets": "10.128.0.0/16/24",
+            "mtu": 1300,
+            "netAttachDefName": "ns1/l3-network",
+            "primaryNetwork": true
```

Comment:
I can add a comment about this so we don't forget to add it to documentation.

Reply:
are we talking about the mp0-0.2/ovn-0.1 addresses? what's the implicitly excluded addresses?

Reply:
@tssurya the default gw address of the subnet is choosen by ovn-k and will not be assigned to the pods.

Reply:
I'm with @qinqon .

Something I now realize is the user can only specify a network as primary **if** that network has a subnet defined, right ? Otherwise, we wouldn't be able to egress.

This should also be documented.